### ISSUE TITLE
Add a warning when wrapping an unwrapped Optional into an Optional

### ISSUE THREAD

**Issue by TheBlckbird** [rust-lang/rust-clippy#12143]
### What it does

It would be pretty unnecessary if you were to unwrap an Optional value just to wrap it into an Optional again. See the example for more context. But only warn about this if nothing else is done with the unwrapped value.

### Advantage

- Less code

### Drawbacks

- The code could become more unreadable, but considering what other lints do to reduce the code, I think this is fine.

### Example

```rust
fn my_fun() -> Option<i32> {
  Some(optional_return_value().unwrap())
}
```

Could be written as:

```rust
fn my_fun() -> Option<i32> {
  optional_return_value()
}
```

**Comment by samueltardieu** [comment]
The first version panics if `optional_return_value()` returns `None`, while the second one just returns `None`). Such a transformation would be invalid.

**Comment by samueltardieu** [comment]
One valid rewrite would be `Some(optional_return_value()?)` into `optional_return_value`, and this is already taken care of by the `needless_question_mark` lint.

**Comment by TheBlckbird** [comment]
There are
